File: WalletKitKotlin/src/commonMain/kotlin/Currency.kt
package com.breadwallet.core

import kotlinx.io.core.Closeable

/** A currency is a medium for exchange. */
expect class Currency : Closeable {

    /** 'A Unique Identifier */
    public val uids: String

    /** The code; e.g. BTC */
    public val code: String

    /** The name; e.g. Bitcoin */
    public val name: String

    /** The type: */
    public val type: String

    /** The issuer, if present.  This is generally an ERC20 address. */
    public val issuer: String?

    override fun equals(other: Any?): Boolean
    override fun hashCode(): Int

    companion object {
        fun create(
                uids: String,
                name: String,
                code: String,
                type: String,
                issuer: String?
        ): Currency
    }
}

/** Used to map Currency -> Built-In-Blockchain-Network */
const val CURRENCY_CODE_AS_BTC = "btc"

/** Used to map Currency -> Built-In-Blockchain-Network */
const val CURRENCY_CODE_AS_BCH = "bch"

/** Used to map Currency -> Built-In-Blockchain-Network */
const val CURRENCY_CODE_AS_ETH = "eth"

File: WalletKitJS/WalletKitCoreWASM/Currency.ts
import CoreJS from "./libcorecrypto.js"

type Currency = number;

const CurrencyInterface = {
    "cryptoCurrencyCreate": CoreJS.cwrap('cryptoCurrencyCreate', 'int', [ 'string', 'string', 'string', 'string', 'string' ]),
    
    "currencyCreate": (uids: string, name: string, code: string, type: string, issuer: string | undefined): Currency => {
	uids; name; code; type; issuer;
	return 1;
    },
    "currencyGetUids": (core: Currency): string => {
	core;
	return 'uids';
    },
};

export {
    Currency,
    CurrencyInterface
}
